core/src/it/scala/org/ensime/indexer/SearchServiceSpec.scala

```diff
@@ -501,32 +512,39 @@ object SearchServiceTestUtils {
     results
   }
 
-  def searchExpectEmpty(query: String)(implicit service: SearchService,
-                                       p: Position) = {
+  def searchExpectEmpty(
+    query: String
+  )(implicit service: SearchService, p: Position, s: Strategy) = {
     val max     = 1
-    val results = service.searchClassesMethods(List(query), max)
+    val results = service.searchClassesMethods(List(query), max).unsafeRun()
     withClue(s"expected empty results from $query")(results shouldBe empty)
     results
   }
 
-  def searchesEmpty(queries: String*)(implicit service: SearchService,
-                                      p: Position) =
+  def searchesEmpty(
+    queries: String*
+  )(implicit service: SearchService, p: Position, strategy: Strategy) =
     queries.toList.foreach(searchExpectEmpty)
 
   // doesn't assert that expect finds itself because the lucene query
   // syntax conflicts with the characters in method FQNs
   def searchesMethods(
     expect: String,
     queries: String*
-  )(implicit service: SearchService, p: Position) =
+  )(implicit service: SearchService, p: Position, strategy: Strategy) =
     (queries.toList).foreach(searchClassesAndMethods(expect, _))
 
   def getClassHierarchy(
     fqn: String,
     hierarchyType: Hierarchy.Direction
-  )(implicit service: SearchService, p: Position): Hierarchy = {
+  )(implicit service: SearchService,
+    p: Position,
+    strategy: Strategy): Hierarchy = {
     val hierarchy =
-      Await.result(service.getTypeHierarchy(fqn, hierarchyType), Duration.Inf)
+      Await.result(
+        service.getTypeHierarchy(fqn, hierarchyType).unsafeRunAsyncFuture(),
```

Comment:
Is there any point in awaiting a future infinitely instead of running it synchronously with `unsafeRun`?

Reply:
Looks pretty much equivalent from fs2's code. I was just trying to minimize introduced changes :-)